### ISSUE TITLE
Change how relations are specified in the ParametricDR class

### ISSUE THREAD

**Issue by einbandi** [einbandi/paradime#2]
Distinguish between

- "global" relations that are applied to the whole dataset (e.g., perplexity-based stuff in t-SNE); and
- "local" relations that are calculated on-the-fly/batch-wise.

The former should be defined once in the beginning (but still have to be accessible by the losses later). The latter should be specified for each training phase. This should make use of the new `TrainingPhase` class or the equivalent `add_training_phase` method. To allow for multiple global relations, use the same key value syntax as already used in the batch accessing.

Users should be able to pass a single global and/or local relation, which will be assigned a default key. The default key is used as a default value for one of the keyword arguments in the pre-defined loss functions (so it can be overidden easily).
